app/role.go

```diff
@@ -59,10 +111,46 @@ func (a *App) UpdateRole(role *model.Role) (*model.Role, *model.AppError) {
 	if err != nil {
 		return nil, err
 	}
-	a.sendUpdatedRoleEvent(savedRole)
 
-	return savedRole, nil
+	builtInChannelRoles := []string{
+		model.CHANNEL_GUEST_ROLE_ID,
+		model.CHANNEL_USER_ROLE_ID,
+		model.CHANNEL_ADMIN_ROLE_ID,
+	}
+
+	builtInRolesMinusChannelRoles := utils.RemoveStringsFromSlice(model.BuiltInSchemeManagedRoleIDs, builtInChannelRoles...)
+
```

Comment:
Just an observation, if we would have Scope on Roles, one could then just use the role scope to filter out the non-Channel scoped roles (you do it in some other place here, where you iterate over all permissions and use their Scope for early return).

Reply:
Agreed. Currently the scope of a role is inferred by what column it is pointed to from one of the `Schemes.Default*Role` columns.  

Those columns are encoding both the _scope_ and the _type_ of the role, like so: `Schemes.Default[Scope][Type]Role` where Scope = `System`|`Team`|`Channel`, and Type = `Admin`|`User`|`Guest`.

It makes the queries much more difficult.

Reply:
having the scope on the role is just one among many things we could do as part of a tech debt roles project that would be helpful when its time to work on "custom" roles